src/NuGet.Core/NuGet.CommandLine.XPlat/Utility/MSBuildAPIUtility.cs

```diff
@@ -120,6 +120,39 @@ public int RemovePackageReference(string projectPath, LibraryDependency libraryD
             }
         }
 
+        /// <summary>
+        /// Check if the project files format are correct for CPM
+        /// </summary>
+        /// <param name="packageReferenceArgs">Arguments used in the command</param>
+        /// <returns></returns>
+        public bool AreCentralVersionRequirementsSatisfied(PackageReferenceArgs packageReferenceArgs)
+        {
+            var project = GetProject(packageReferenceArgs.ProjectPath);
+
+            // Get package version and VersionOverride if it already exists in the props file. Returns null if there is no matching package version.
+            ProjectItem packageReferenceInProps = project.Items.LastOrDefault(i => i.ItemType == PACKAGE_REFERENCE_TYPE_TAG && i.EvaluatedInclude.Equals(packageReferenceArgs.PackageId));
+            var versionOverrideExists = packageReferenceInProps?.Metadata?.FirstOrDefault(i => i.Name.Equals("VersionOverride"));
+
+            // Get package version if it already exists in the props file. Returns null if there is no matching package version.
+            ProjectItem packageVersionInProps = project.Items.LastOrDefault(i => i.ItemType == PACKAGE_VERSION_TYPE_TAG && i.EvaluatedInclude.Equals(packageReferenceArgs.PackageId));
```

Comment:
We'll need to differentiate between `<PackageVersion />` items in the project versus `Directory.Packages.props`.  I think we'll need to check the `ContainingProject` of the item to make sure things are configured correctly.  Ping me to discuss if you need more info